htlcswitch/interfaces.go

```diff
@@ -137,7 +137,7 @@ type ChannelUpdateHandler interface {
 	// ShutdownIfChannelClean shuts the link down if the channel state is
 	// clean. This can be used with dynamic commitment negotiation or coop
 	// close negotiation which require a clean channel state.
-	ShutdownIfChannelClean() error
+	ShutdownHtlcManager()
```

Comment:
`godoc` comment not updated.

Reply:
But if we have the new flush call, then in situations where's a flush to shut down, can't we just add that context to an arg in `Flush`. This way the caller doesn't need to reach down as much into the implementation details of how the link works. Instead if the extra arg is set, then the link starts to down itself down when it's safe to do so. 

Technically, the link should still be "up" (but in the fully flushed state) until shutdown is sent and recv'd by other parties. As there's a provision that lets either side send a new `update_fee` to adjust the current fee level.